### PR TITLE
Replace bookmark manager empty list string

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/2494

## Submitter Checklist:
Screen Shot 2020-01-10 at 10 50 49 AM

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: patches/chrome-browser-ui-webui-bookmarks-bookmarks_ui.cc.patch [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 --- a/chrome/browser/ui/webui/bookmarks/bookmarks_ui.cc
 +++ b/chrome/browser/ui/webui/bookmarks/bookmarks_ui.cc
-@@ -46,6 +46,7 @@ content::WebUIDataSource* CreateBookmarksUIHTMLSource(Profile* profile) {
-   content::WebUIDataSource* source =
-       content::WebUIDataSource::Create(chrome::kChromeUIBookmarksHost);
+@@ -174,6 +174,7 @@ content::WebUIDataSource* CreateBookmarksUIHTMLSource(Profile* profile) {
+ 
+   source->UseStringsJs();
  
 +  BraveCustomizeBookmarksDataSource(source);
```
Comment (simonhong) [inline patches/chrome-browser-ui-webui-bookmarks-bookmarks_ui.cc.patch]: Moved `BraveCustomizeBookmarksDataSource(source)` to the last of `CreateBookmarksUIHTMLSource()` to get a chance to replace already added strings.

**Comment by simonhong** [comment]
Only unrelated steps (`network-audit` and `test-install`) were failed.

File: patches/chrome-browser-ui-webui-bookmarks-bookmarks_ui.cc.patch (hunk 1 above)
Comment (petemill) [reply patches/chrome-browser-ui-webui-bookmarks-bookmarks_ui.cc.patch]: 👍

**Comment by simonhong** [comment]
@bridiver ping for owners review :)

File: patches/chrome-browser-ui-webui-bookmarks-bookmarks_ui.cc.patch (hunk 1 above)
Comment (bridiver) [reply patches/chrome-browser-ui-webui-bookmarks-bookmarks_ui.cc.patch]: can you convert this to a `define` since you're changing it anyway?

Comment (simonhong) [reply patches/chrome-browser-ui-webui-bookmarks-bookmarks_ui.cc.patch]: done.

**Pull request merged**